Entities/AEmitter.cpp

```diff
@@ -482,46 +478,51 @@ void AEmitter::Update()
 
                     // Now figure how many full emissions can fit in the current accumulator
-                    emissions = floor((*eItr)->m_Accumulator / SPE);
+					emissionCount = std::floor((*eItr)->m_Accumulator / SPE);
                     // Deduct the about to be emitted emissions from the accumulator
-                    (*eItr)->m_Accumulator -= emissions * SPE;
+                    (*eItr)->m_Accumulator -= emissionCount * SPE;
 
                     RTEAssert((*eItr)->m_Accumulator >= 0, "Emission accumulator negative!");
                 }
-
+				else {
+					(*eItr)->m_Accumulator = 0;
+				}
+				float scale = 1.0F;
                 // Add extra emissions if bursting.
-                if (m_BurstTriggered)
-                    emissions += (*eItr)->GetBurstSize();
-
+				if (m_BurstTriggered) {
+					emissionCount += (*eItr)->GetBurstSize() * throttleFactor;
+					scale = m_BurstScale;
+				}
+				emissionCountTotal += emissionCount;
                 pParticle = 0;
                 emitVel.Reset();
                 parentVel = pRootParent->GetVel() * (*eItr)->InheritsVelocity();
 
-                for (int i = 0; i < emissions; ++i)
+                for (int i = 0; i < emissionCount; ++i)
                 {
-                    velMin = (*eItr)->GetMinVelocity() * (m_BurstTriggered ? m_BurstScale : 1.0);
-                    velRange = (*eItr)->GetMaxVelocity() - (*eItr)->GetMinVelocity() * (m_BurstTriggered ? m_BurstScale : 1.0);
-                    spread = (*eItr)->GetSpread() * (m_BurstTriggered ? m_BurstScale : 1.0);
+                    velMin = (*eItr)->GetMinVelocity() * scale;
+                    velRange = (*eItr)->GetMaxVelocity() - (*eItr)->GetMinVelocity() * scale;
+                    spread = (*eItr)->GetSpread() * scale;
                     // Make a copy after the reference particle
                     pParticle = dynamic_cast<MovableObject *>((*eItr)->GetEmissionParticlePreset()->Clone());
                     // Set up its position and velocity according to the parameters of this.
                     // Emission point offset not set
 
-					if ((*eItr)->GetOffset().IsZero())
-					{
-						if (m_EmissionOffset.IsZero())
-							pParticle->SetPos(m_Pos/*Vector(m_Pos.m_X + 5 * NormalRand(), m_Pos.m_Y + 5 * NormalRand())*/);
-						else
+					if ((*eItr)->GetOffset().IsZero()) {
+						if (m_EmissionOffset.IsZero()) {
+							pParticle->SetPos(m_Pos);
+						} else {
 							pParticle->SetPos(m_Pos + RotateOffset(m_EmissionOffset));
-					}
-					else 
-					{
+						}
+					} else {
 						pParticle->SetPos(m_Pos + RotateOffset((*eItr)->GetOffset()));
 					}
     // TODO: Optimize making the random angles!")
                     emitVel.SetXY(velMin + RandomNum(0.0F, velRange), 0.0F);
 					emitVel.RadRotate(m_EmitAngle.GetRadAngle() + spread * RandomNormalNum());
                     emitVel = RotateOffset(emitVel);
+					// Scale particle velocity based on throttle?
+					//emitVel *= throttleFactor;
```

Comment:
If you wanna keep this as a reminder for future, please stick in a TODO
Also, I feel like that may make things complicated, but I'm not really sure.